ethers/provider.nim

```diff
@@ -17,13 +17,40 @@ type
   Log* = object
     data*: seq[byte]
     topics*: seq[Topic]
+  TransactionHash* = array[32, byte]
+  BlockHash* = array[32, byte]
+  TransactionStatus* = enum
+    Failure = 0,
+    Success = 1,
+    Invalid = 2
+  TransactionResponse* = object
+    provider*: Provider
+    hash*: TransactionHash
+  TransactionReceipt* = object
+    sender*: ?Address
+    to*: ?Address
+    contractAddress*: ?Address
+    transactionIndex*: UInt256
+    gasUsed*: UInt256
+    logsBloom*: seq[byte]
+    blockHash*: BlockHash
+    transactionHash*: TransactionHash
+    logs*: seq[Log]
+    blockNumber*: ?UInt256
+    cumulativeGasUsed*: UInt256
+    status*: TransactionStatus
   LogHandler* = proc(log: Log) {.gcsafe, upraises:[].}
+  BlockHandler* = proc(blck: Block) {.gcsafe, upraises:[].}
   Topic* = array[32, byte]
   Block* = object
     number*: UInt256
     timestamp*: UInt256
     hash*: array[32, byte]
 
+const DEFAULT_CONFIRMATIONS* {.intdefine.} = 12
```

Comment:
Updated. I don't think `{.intdefine.}` is the best option in the end (perhaps some kind of in-memory settings would be better suited), but I'd prefer to provide at least compile-time control over these values to consumer projects instead of completely hardcoding them.